[CI] ResolveClusterActionResponseTests testEqualsAndHashcode failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/o4dwpco7p25fw/tests/:server:test/org.elasticsearch.action.admin.indices.resolve.ResolveClusterActionResponseTests/testEqualsAndHashcode


**Reproduction line:**
```
./gradlew ':server:test' --tests "org.elasticsearch.action.admin.indices.resolve.ResolveClusterActionResponseTests.testEqualsAndHashcode" -Dtests.seed=E7520D27252DAE83 -Dtests.locale=el -Dtests.timezone=Indian/Chagos -Druntime.java=21
```

**Applicable branches:**
8.13

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.action.admin.indices.resolve.ResolveClusterActionResponseTests#testEqualsAndHashcode`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testEqualsAndHashcode'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.action.admin.indices.resolve.ResolveClusterActionResponseTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
java.lang.AssertionError: ResolveClusterActionResponse mutation should not be equal to original
Expected: not <org.elasticsearch.action.admin.indices.resolve.ResolveClusterActionResponse@47fbffe4>
     but: was <org.elasticsearch.action.admin.indices.resolve.ResolveClusterActionResponse@47fbffe4>

  at __randomizedtesting.SeedInfo.seed([E7520D27252DAE83:965D75EAEACAE7AC]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:964)
  at org.elasticsearch.test.EqualsHashCodeTestUtils.checkEqualsAndHashCode(EqualsHashCodeTestUtils.java:95)
  at org.elasticsearch.test.AbstractWireTestCase.testEqualsAndHashcode(AbstractWireTestCase.java:60)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)

```

--- a/server/src/test/java/org/elasticsearch/action/admin/indices/resolve/ResolveClusterActionResponseTests.java
+++ b/server/src/test/java/org/elasticsearch/action/admin/indices/resolve/ResolveClusterActionResponseTests.java
@@ -15,35 +15,41 @@ import org.elasticsearch.transport.RemoteClusterAware;
 
 import java.util.HashMap;
 import java.util.Map;
-import java.util.function.Supplier;
 
 public class ResolveClusterActionResponseTests extends AbstractWireSerializingTestCase<ResolveClusterActionResponse> {
 
     @Override
     protected ResolveClusterActionResponse createTestInstance() {
-        return new ResolveClusterActionResponse(randomResolveClusterInfoMap(() -> randomResolveClusterInfo()));
+        return new ResolveClusterActionResponse(randomResolveClusterInfoMap(null));
     }
 
-    private Map<String, ResolveClusterInfo> randomResolveClusterInfoMap(Supplier<ResolveClusterInfo> rcInfoSupplier) {
+    private ResolveClusterInfo randomResolveClusterInfo(ResolveClusterInfo existing) {
+        if (existing == null) {
+            return randomResolveClusterInfo();
+        } else {
+            return randomValueOtherThan(existing, () -> randomResolveClusterInfo());
+        }
+    }
+
+    private ResolveClusterInfo getResolveClusterInfoFromResponse(String key, ResolveClusterActionResponse response) {
+        if (response == null || response.getResolveClusterInfo() == null) {
+            return null;
+        }
+        return response.getResolveClusterInfo().get(key);
+    }
+
+    private Map<String, ResolveClusterInfo> randomResolveClusterInfoMap(ResolveClusterActionResponse existingResponse) {
         Map<String, ResolveClusterInfo> infoMap = new HashMap<>();
-        int numClusters = randomIntBetween(0, 100);
+        int numClusters = randomIntBetween(0, 50);
         if (randomBoolean() || numClusters == 0) {
-            infoMap.put(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY, rcInfoSupplier.get());
+            String key = RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY;
+            infoMap.put(key, randomResolveClusterInfo(getResolveClusterInfoFromResponse(key, existingResponse)));
         }
         for (int i = 0; i < numClusters; i++) {
-            infoMap.put("remote_" + i, rcInfoSupplier.get());
+            String key = "remote_" + i;
+            infoMap.put(key, randomResolveClusterInfo(getResolveClusterInfoFromResponse(key, existingResponse)));
         }
         return infoMap;
-    }
-
-    /**
-     * Useful for the mutateInstance method to ensure that no the new ResolveClusterInfo will not
-     * match the original one.
-     *
-     * @return version of ResolveClusterInfo that has randomized error strings
-     */
-    static ResolveClusterInfo randomResolveClusterInfoWithErrorString() {
-        return new ResolveClusterInfo(randomBoolean(), randomBoolean(), randomAlphaOfLength(125));
     }
 
     static ResolveClusterInfo randomResolveClusterInfo() {
@@ -63,6 +69,6 @@ public class ResolveClusterActionResponseTests extends AbstractWireSerializingTe
 
     @Override
     protected ResolveClusterActionResponse mutateInstance(ResolveClusterActionResponse response) {
-        return new ResolveClusterActionResponse(randomResolveClusterInfoMap(() -> randomResolveClusterInfoWithErrorString()));
+        return new ResolveClusterActionResponse(randomResolveClusterInfoMap(response));
     }
 }

--- a/server/src/test/java/org/elasticsearch/action/admin/indices/resolve/ResolveClusterActionResponseTests.java
+++ b/server/src/test/java/org/elasticsearch/action/admin/indices/resolve/ResolveClusterActionResponseTests.java
@@ -15,27 +15,38 @@ import org.elasticsearch.transport.RemoteClusterAware;
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.function.Supplier;
 
 public class ResolveClusterActionResponseTests extends AbstractWireSerializingTestCase<ResolveClusterActionResponse> {
 
     @Override
     protected ResolveClusterActionResponse createTestInstance() {
-        return new ResolveClusterActionResponse(randomResolveClusterInfoMap());
+        return new ResolveClusterActionResponse(randomResolveClusterInfoMap(() -> randomResolveClusterInfo()));
     }
 
-    private Map<String, ResolveClusterInfo> randomResolveClusterInfoMap() {
+    private Map<String, ResolveClusterInfo> randomResolveClusterInfoMap(Supplier<ResolveClusterInfo> rcInfoSupplier) {
         Map<String, ResolveClusterInfo> infoMap = new HashMap<>();
-        int numClusters = randomIntBetween(0, 50);
+        int numClusters = randomIntBetween(0, 100);
         if (randomBoolean() || numClusters == 0) {
-            infoMap.put(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY, randomResolveClusterInfo());
+            infoMap.put(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY, rcInfoSupplier.get());
         }
         for (int i = 0; i < numClusters; i++) {
-            infoMap.put("remote_" + i, randomResolveClusterInfo());
+            infoMap.put("remote_" + i, rcInfoSupplier.get());
         }
         return infoMap;
     }
 
-    private ResolveClusterInfo randomResolveClusterInfo() {
+    /**
+     * Useful for the mutateInstance method to ensure that no the new ResolveClusterInfo will not
+     * match the original one.
+     *
+     * @return version of ResolveClusterInfo that has randomized error strings
+     */
+    static ResolveClusterInfo randomResolveClusterInfoWithErrorString() {
+        return new ResolveClusterInfo(randomBoolean(), randomBoolean(), randomAlphaOfLength(125));
+    }
+
+    static ResolveClusterInfo randomResolveClusterInfo() {
         int val = randomIntBetween(1, 3);
         return switch (val) {
             case 1 -> new ResolveClusterInfo(false, randomBoolean());
@@ -52,6 +63,6 @@ public class ResolveClusterActionResponseTests extends AbstractWireSerializingTe
 
     @Override
     protected ResolveClusterActionResponse mutateInstance(ResolveClusterActionResponse response) {
-        return new ResolveClusterActionResponse(randomResolveClusterInfoMap());
+        return new ResolveClusterActionResponse(randomResolveClusterInfoMap(() -> randomResolveClusterInfoWithErrorString()));
     }
 }

